src/components/file/UploadFile.tsx

```diff
@@ -37,49 +27,59 @@ const UploadFile = (): JSX.Element => {
 	const uploadFile = async () => {
 		if (!fileEvent) return;
 		setIsLoading(true);
-		const filename = extractFilename(fileEvent.target.value);
+		const fileName = extractFilename(fileEvent.target.value);
 		const fileContent = await getFileContent(fileEvent.target.files ? fileEvent.target.files[0] : []);
-		const key = generateFileKey();
-		if (!filename || !fileContent) {
+
+		if (!fileName || !fileContent) {
 			toast({ title: 'Invalid file', status: 'error' });
 			setFileEvent(undefined);
 			setIsLoading(false);
-			return;
-		}
+		} else {
+			const cryptoKey = await crypto.subtle.generateKey(
+				{
+					name: 'AES-GCM',
+					length: 256,
+				},
+				true,
+				['encrypt', 'decrypt'],
+			);
+			const keyString = await crypto.subtle.exportKey('raw', cryptoKey);
+			const iv = crypto.getRandomValues(new Uint8Array(128));
 
-		const file: IPCFile = {
-			id: uuid(),
-			name: filename,
-			hash: fileContent,
-			size: fileEvent.target.files![0].size,
-			createdAt: Date.now(),
-			key: { iv: '', ephemPublicKey: '', ciphertext: '', mac: '' },
-			path,
-			deletedAt: null,
-			permission: 'owner',
-			logs: [{
-				action: "File created",
-				date: Date.now()
-			}],
-		};
+			const file: IPCFile = {
+				id: uuid(),
```

Comment:
Now we could also use the `crypto.randomUUID()` built-in API method and remove the `uuid` dependency :wink: